src/NuGet.Core/NuGet.Commands/RestoreCommand/CompatibilityChecker.cs

```diff
@@ -243,17 +243,21 @@ private static RestoreLogMessage GetErrorMessage(NuGetLogCode logCode, Compatibi
             {
                 foreach (var group in contentItems.FindItemGroups(pattern))
                 {
-                    object tfmObj = null;
-                    object ridObj = null;
-                    group.Properties.TryGetValue(ManagedCodeConventions.PropertyNames.RuntimeIdentifier, out ridObj);
-                    group.Properties.TryGetValue(ManagedCodeConventions.PropertyNames.TargetFrameworkMoniker, out tfmObj);
+                    // lib/net45/subfolder/a.dll will be returned as a group with zero items since sub
+                    // folders are not allowed. Completely empty groups are not compatible, a group with
+                    // _._ would contain _._ as an item.
+                    if (group.Items.Count > 0)
```

Comment:
Just a suggestion - you could reduce nesting with a `continue` instead :)

Reply:
I appreciate the suggestion, it could work that way. I usually favor the single exit point strategy over reducing nesting unless it gets too crazy. Either one is probably fine here :smile: